### PR TITLE
OPENDNSSEC-790: align worker code base

### PR DESCRIPTION
**Pull request by halderen**

### CODE REVIEW

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
Bind named on redhat dumps core:
core.13011: ELF 64-bit LSB core file x86-64, version 1 (SYSV), SVR4-style, from 'named -p 10053 -c /home/jenkins/workspace/pull-opendnssec-594-test/label/redhat'
Unable to fix test because of this, and the version on RH is a release candidate that is unable to update due to RH being out of LTS.

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
@yschaeff @hodarh Please review and merge.  Also contains fix for current daily build being broken.  Still reviewing of this fixes also the "now" problem and enforce task "skip".

File: signer/src/signer/zone.c [hunk 1]
```diff
@@ -1083,6 +1084,7 @@ zone_recover2(zone_type* zone)
                 "create task", zone_str, zone->name);
             goto recover_error2;
         }
+        task->lock = &zone->zone_lock;
```
Comment (yschaeff) [inline signer/src/signer/zone.c]: This makes me nervous. Following tasks MAY not get the same lock. At least in theory. It is also a recipe for deadlock when trying to grab zone->zone_lock inside an executed task.Is this safe?

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
I hear all these wonderfull things about git and automerging never going wrong, but in practise git always puts up weird and wonderful bad stuff:

+ git merge d22ab5fa43f357df220dbca36f0042c25d622b38
Removing testing/test-cases-daily.d/signer.adaptors.bind_stress_test/bind9/ods.bak
Auto-merging signer/src/daemon/signertasks.h
Auto-merging signer/src/daemon/signertasks.c
Auto-merging signer/src/daemon/engine.c
CONFLICT (content): Merge conflict in signer/src/daemon/engine.c
Auto-merging common/scheduler/worker.h
Auto-merging common/scheduler/worker.c
Auto-merging common/scheduler/schedule.c
Auto-merging common/scheduler/fifoq.h
Auto-merging common/scheduler/fifoq.c
Automatic merge failed; fix conflicts and then commit the result.

In fact both git locally and on github tell me that it is up-to-date with upstream/develop.
So I'm administratively pulling in this merge coz there is no other way to merge it anymore.

**Pull request merged**